src/Compilers/Core/Portable/InternalUtilities/NoThrowStream.cs

```diff
@@ -0,0 +1,31 @@
+﻿
+using System;
+using System.IO;
+
+namespace Roslyn.Utilities
+{
+    struct NoThrowStreamDisposer : IDisposable
+    {
+        public Stream Stream { get; }
+
+        private readonly Action<Exception> _exceptionHandler;
+        
+        public NoThrowStreamDisposer(Stream stream, Action<Exception> exceptionHandler)
```

Comment:
> Action<Exception> exceptionHandler [](start = 51, length = 35)

Do we need to make this so generic? It seems that the lambdas passed to all usages are pretty much the same. I would prefer we avoid lambdas to make the use sites a bit more readable.